src/NServiceBus.Core/Unicast/Subscriptions/MessageDrivenSubscriptions/SubscriptionManager.cs

```diff
@@ -65,17 +65,19 @@ void SendSubscribeMessageWithRetries(Address destination, TransportMessage subsc
                     ReplyToAddress = Configure.PublicReturnAddress
                 });
             }
-            catch (QueueNotFoundException ex)
+            catch (Exception ex)
             {
-                if (retriesCount < 10)
+                if (ex is QueueNotFoundException)
                 {
-                    Thread.Sleep(TimeSpan.FromSeconds(2));
-                    SendSubscribeMessageWithRetries(destination, subscriptionMessage, messageType, ++retriesCount);
-                }
-                else
-                {
-                    Logger.ErrorFormat("Failed to subscribe to {0} at publisher queue {1}", ex, messageType, destination);
+                    if (retriesCount < 10)
+                    {
+                        Thread.Sleep(TimeSpan.FromSeconds(2));
+                        SendSubscribeMessageWithRetries(destination, subscriptionMessage, messageType, ++retriesCount);
+                        return;
+                    }
                 }
+
+                Logger.FatalFormat("Failed to subscribe to {0} at publisher queue {1}. Error: {2}", messageType, destination, ex);
```

Comment:
A fatal error usually means a process stop. I.e. it is better for the process to stop.

Is this really a fatal error? If so, would we prefer the process to stop?